ui/components/app/alert-system/alert-modal/alert-modal.tsx

```diff
@@ -311,10 +314,14 @@ export function AlertModal({
         />
         <AlertHeader selectedAlert={selectedAlert} customTitle={customTitle} />
         <ModalBody>
-          <AlertDetails
-            selectedAlert={selectedAlert}
-            customDetails={customDetails}
-          />
+          {selectedAlert?.provider === SecurityProvider.Blockaid ? (
+            <BlockaidAlertDetails />
```

Comment:
Why does this need to be hardcoded inside the `AlertModal` which aims to be decoupled from any single alert or domain?

Could the `blockaidAlertInfo` label instead be used when the alert is generated from the relevant `useConfirmationAlerts` hook so it's not a special case, but a standard alert like all others?

Reply:
Please check this: https://github.com/MetaMask/metamask-extension/pull/24702#discussion_r1618805115